src/scripts/components/Chart.jsx

```diff
@@ -43,7 +52,7 @@ const Chart = (props) => {
           </div>
         )}
       </div>
-      {renderChart(opts)}
+      { (typeof opts.data === 'undefined' || opts.data === null || Object.keys(opts.data).length === 0) ? renderNoData() : renderChart(opts) }
```

Comment:
Is there a need to check for all these? Are you not able to determine what the value of data is when it's empty? I'm curious if sometimes it's `undefined` and sometimes it's `null` or if you're just checking for both because you're not sure. I would rather we understand the shape of the data and check specifically for the empty state.

Reply:
`{
      labels: ['January', 'February', 'March', 'April', 'May', 'June'],
      datasets: [
        {
          label: 'Message Count',
          backgroundColor: 'rgba(210,235,248,0.5)',
          borderColor: 'rgba(36,133,207,1)',
          borderWidth: 1,
          hoverBackgroundColor: 'rgba(255,99,132,0.4)',
          hoverBorderColor: 'rgba(255,99,132,1)',
          data: [65, 59, 80, 81, 56, 55]
        },
      ],
    }`

The data object comes in the above format and we want to make sure it passes all of the cases when the data is undefined, null or empty. We might also need to check on another condition where there is no data in data.datasets[0].data. @alexgsummer

Reply:
OK, I see. This is handling multiple data formats. Thanks for clarification.